pennylane/_qubit_device.py

```diff
@@ -112,83 +112,237 @@ def execute(self, queue, observables, parameters=None):
 
             # Ensures that a combination with sample does not put
             # expvals and vars in superfluous arrays
-            if all(obs.return_type is Sample for obs in observables):
-                return self._asarray(results)
-            if any(obs.return_type is Sample for obs in observables):
+            sample_return_types = (obs.return_type is Sample for obs in observables)
+            if any(sample_return_types) and not all(sample_return_types):
                 return self._asarray(results, dtype="object")
 
             return self._asarray(results)
 
-    def rotate_basis(self, observable):
+    def statistics(self, observables):
+        """Process measurement results from circuit execution and return statistics.
+
+        This includes returning expectation values, variance, samples and probabilities.
+
+        Args:
+            observables (Union[:class:`Observable`, List[:class:`Observable`]]): the number of basis states to sample from
+
+        Raises:
+            QuantumFunctionError: if the value of :attr:`~.Observable.return_type` is not supported
+
+        Returns:
+            Union[float, List[float]]: the corresponding statistics
+        """
+        results = []
+
+        for obs in observables:
+            # Pass instances directly
+            if obs.return_type is Expectation:
+                results.append(self.expval(obs))
+
+            elif obs.return_type is Variance:
+                results.append(self.var(obs))
+
+            elif obs.return_type is Sample:
+                results.append(np.array(self.sample(obs)))
+
+            elif obs.return_type is Probability:
+                results.append(self.probability(wires=obs.wires))
+
+            elif obs.return_type is not None:
+                raise QuantumFunctionError("Unsupported return type specified for observable {}".format(obs.name))
+
+        return results
+
+    def rotate_basis(self, obs_queue):
         """Rotates the specified wires such that they
         are in the eigenbasis of the provided observable.
 
         Args:
-            observable (Observable): the observable that is to be
-                measured
+            observables (List[:class:`Observable`]): the observable queue
+        """
+        self._prob = self.probability()
```

Comment:
This doesn't seem to be used anywhere :thinking:

Reply:
Correct! This one was added such that the probability before the rotation is stored and can be potentially retrieved

Reply:
What was the use case you had in mind for this pre-rotation probability?

Reply:
The notion behind this was to be able to perform sanity checks by looking at the circuits. Ultimately it might be better to store the state in fashion as ``dev._state`` does

Reply:
Yep, I agree. Probability can now be returned via the QNode, so it makes more sense to cache the pre-rotation state.

Reply:
Okay, I see. But is this something intended for internal use only or is there a special UI for accessing the probabilities pre-rotation?

Reply:
I think this is something that it outdated, and will be changed to instead cache the *state* pre rotation.